app/models/user.rb

```diff
@@ -0,0 +1,11 @@
+class User < ApplicationRecord
+  has_many :created_events, foreign_key: 'creator_id', class_name: 'Event'
+  has_many :invitations, foreign_key: :attendee_id
+  has_many :attended_events, through: :invitations, source: :attended_event
+
+  validates :name, presence: true, uniqueness: true, length: { minimum: 4 }
+  validates :email, presence: true, uniqueness: true, length: { minimum: 11 }
+
+  scope :past, -> { where('date < ?', Date.today) }
+  scope :future, -> { where('date >= ?', Date.today) }
```

Comment:
Why this?

Reply:
Scope was required for past and upcoming events

Reply:
Yes but only for `Event`.
If you check your code, you will notice that you didn't use it anywhere. You used the one from `Event` model.

Reply:
ohhh...very true.
Thank you for this